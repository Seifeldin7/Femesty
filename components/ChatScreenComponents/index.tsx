export * from './ChatInputBox'
export * from './ChatMessage'