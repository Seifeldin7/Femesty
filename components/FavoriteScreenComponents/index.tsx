export * from "./FavoriteHzScroller";
export * from "./FavoriteLook";
