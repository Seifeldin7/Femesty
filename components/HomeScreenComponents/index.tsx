export * from "./FilterButton"
export * from "./HeaderTab"
export * from "./HorizontalScroller"
export * from "./LookMedia"
export * from "./SideBar"