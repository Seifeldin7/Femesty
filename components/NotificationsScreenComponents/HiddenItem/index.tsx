import { AntDesign } from "@expo/vector-icons";
import React from "react";
import {
  View,
  Text,
  TouchableHighlight,
} from "react-native";
import { responsiveHeight } from "react-native-responsive-dimensions";

import { styles } from "./styles";
import { componentProps } from "./types";

export const HiddenItem = ({ data, onMarkAsRead }: componentProps) => {
  const height = data.item?.subject_type == "Order" ? responsiveHeight(13) : responsiveHeight(13);
  return (
    <View style={{ ...styles.rowBack, height: height }}>
      <TouchableHighlight
        style={{ ...styles.backRightBtn, height: height }}
        onPress={() => onMarkAsRead(data.item.id)}
      >
        <View style={{ justifyContent: "center", alignItems: "center" }}>
          <AntDesign
            style={styles.check}
            name="checkcircleo"
            size={26}
            color="black"
          />
          <Text style={{ color: "white" }}>Mark as read</Text>
        </View>
      </TouchableHighlight>
    </View>
  );
};
