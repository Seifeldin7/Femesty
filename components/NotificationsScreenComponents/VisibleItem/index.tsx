import React from "react";
import { View, Text } from "react-native";
import { styles } from "./styles";
import { componentProps } from "./types";
import { Avatar } from "../../ThemeComponents/Avatar";
import { responsiveHeight } from "react-native-responsive-dimensions";

export const VisibleItem = ({ data }: componentProps) => {
  const height = data.item.subject_type == "Order" ? responsiveHeight(13) : responsiveHeight(13);
  return (
    <View style={{ ...styles.rowFront, height: height }}>
      <View style={styles.rowFrontVisible}>
        {data.item.is_read ? (
          <View testID="notificationCircle" style={styles.circle}>
          </View>
        ) : null}
        {data.item.subject_type == "Order" ? (
          <View style={{ flexDirection: "row" }}>
            <Avatar imgPath={{ uri: data.item.subject?.product.image1 }} style={{marginRight: 10}} size="medium" />
            <View>
              <Text style={styles.desc}> {data.item.description}</Text>
              <Text style={styles.productName}>
                {" "}
                {data.item.subject?.product.name}
              </Text>
              {data.item.created_at ? (
                <View style={{ flexDirection: "row" }}>
                  <Text style={styles.productName}> Delivered By: </Text>
                  <Text style={styles.date}> {data.item.created_at}</Text>
                </View>
              ) : null}
            </View>
          </View>
        ) : (
          <View style={{ flexDirection: "row" }}>
            <Avatar imgPath={{ uri: data.item.subject?.image1 }} size="medium" />
            <View>
              <Text style={styles.desc}> {data.item.description}</Text>
              <Text style={styles.productName}> {data.item.subject?.name}</Text>
              {data.item.created_at ? (
                <View style={{ flexDirection: "row" }}>
                  <Text style={styles.productName}> Delivered By: </Text>
                  <Text style={styles.date}> {data.item.created_at}</Text>
                </View>
              ) : null}
            </View>
          </View>
        )}
      </View>
    </View>
  );
};
