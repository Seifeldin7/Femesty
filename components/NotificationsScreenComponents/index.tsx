export * from "./HiddenItem";
export * from "./VisibleItem";
