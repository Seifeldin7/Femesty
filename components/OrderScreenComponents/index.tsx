export * from './OrderItem'
export * from './OrderCard'
export * from './OrderList'