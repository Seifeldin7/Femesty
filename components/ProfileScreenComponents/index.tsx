export * from "./ProfileInformation"
export * from "./MenuList"