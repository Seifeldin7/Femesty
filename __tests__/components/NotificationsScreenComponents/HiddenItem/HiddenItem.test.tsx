import React from "react";
import { fireEvent, render } from "react-native-testing-library";
import { ReactTestInstance } from "react-test-renderer";
import { HiddenItem } from "../../../../components/NotificationsScreenComponents";

test('Should render hidden item correctly in notifications', () => {
    let markAsRead = jest.fn();
    let data = {}
    const tree = render(
        <HiddenItem onMarkAsRead= {markAsRead} data={data}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test("markAsRead function should be called", ()=> {
    let getByText: (text: string) => ReactTestInstance | null;
    let markAsRead = jest.fn();
    let data = {item: {
        id: 1
    }};
    ({ getByText } = render(
        <HiddenItem onMarkAsRead= {markAsRead} data={data}/>
    ))
    const markAsReadBtn = getByText("Mark as read");
    expect(markAsReadBtn).toBeDefined();
    if (markAsReadBtn) {
        fireEvent.press(markAsReadBtn);
        expect(markAsRead).toBeCalledWith(1);
    }
});
