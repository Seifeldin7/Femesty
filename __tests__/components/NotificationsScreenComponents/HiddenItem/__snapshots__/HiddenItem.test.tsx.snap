// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render hidden item correctly in notifications 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "flex": 1,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-between",
      "margin": 5,
      "marginBottom": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8C9DB3",
        "borderBottomRightRadius": 10,
        "borderTopRightRadius": 10,
        "bottom": 0,
        "height": 80,
        "justifyContent": "center",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 150,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Mark as read
      </Text>
    </View>
  </View>
</View>
`;
